### DISCUSSION TITLE
usage of (multiple) atomWithStorage broken in next.js 14.1.0

### DISCUSSION THREAD

**Discussion by therawk** [pmndrs/jotai#2403] [Bug report]
So I just updated next from 14.0.4 to 14.1.0 and noticed that I'm unable to use multiple `atomWithStorage`s.
I keep my atoms in a separate file and when I change a value of one of the storage atoms, the one last `atomWithStorage` will be updated and therefore overwritten by a value that's supposed to be written to a different atom.

If you need a repo/stackblitz let me know but since 14.1.0 is the last version, just give it a try - add 2 `atomWithStorage` and try to update it

**Comment by dai-shi** [comment]
Can you try other Next.js versions? Is it only failing with 14.1.0?
So, I wonder if it's Jotai issue or Next.js issue.
If it's Jotai issue, it should be reproducible without Next.js.
If it's Next.js issue, it should be reproducible without Jotai.

**Comment by therawk** [comment]
I tried it with 14.1.1-canary.15 and it also happens there.
Couldn't test higher versions because they were failing probably because they introduced turbopack as default packager but yeah. Seems like it started witht 14.1.0. -> 14.0.2 also worked

**Comment by dai-shi** [comment]
https://github.com/vercel/next.js/releases/tag/v14.1.0
Hmm, it's hard to tell which has affected.

**Reply by therawk** [reply]
Indeed. The only thing that I really noticed was that while I was trying other version of next, an error popped up regarding jotai and .esm.
I'll try to reproduce this tomorrow and paste the error
